tests/rule_based_profiler/parameter_builder/test_parameter_builder.py

```diff
@@ -0,0 +1,485 @@
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+        batch_request=None,
+        runtime_configuration=None,
+    )
+    assert my_evaluation_dependency_1_parameter_builder.call_count == 1
+
+    """
+    Instead of defining constant list of known fully qualified parameter names, here,
+    "get_fully_qualified_parameter_names()" is used as utility in order to faithfully track actual contents of shared
+    memory as ParameterBuilders are executed; otherwise, initial state of shared memory would not be known exactly.
+    """
+    dependencies_fully_qualified_parameter_names: List[
+        str
+    ] = get_fully_qualified_parameter_names(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+    )
+
+    """
+    These assertions show that both "ParameterBuilder" dependencies computed their values (and stored in shared memory).
+    """
+    # First evaluation "ParameterBuilder" dependency computed its value, and it was stored in shared memory.
+    assert (
+        my_evaluation_dependency_0_parameter_builder.raw_fully_qualified_parameter_name
+        in dependencies_fully_qualified_parameter_names
+        and my_evaluation_dependency_0_parameter_builder.json_serialized_fully_qualified_parameter_name
+        in dependencies_fully_qualified_parameter_names
+    )
+    # Second evaluation "ParameterBuilder" dependency computed its value, and it was stored in shared memory.
+    assert (
+        my_evaluation_dependency_1_parameter_builder.raw_fully_qualified_parameter_name
+        in dependencies_fully_qualified_parameter_names
+        and my_evaluation_dependency_1_parameter_builder.json_serialized_fully_qualified_parameter_name
+        in dependencies_fully_qualified_parameter_names
+    )
+
+    # Now execute dependent "ParameterBuilder".
+    my_dependent_parameter_builder.build_parameters(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+        batch_request=None,
+        runtime_configuration=None,
+    )
+    assert my_dependent_parameter_builder.call_count == 1
+
+    # Assert that precomuted evaluation parameters are not recomputed (call_count for each should remain at 1).
+    assert my_evaluation_dependency_0_parameter_builder.call_count == 1
+    assert my_evaluation_dependency_1_parameter_builder.call_count == 1
+
+    all_fully_qualified_parameter_names: List[
+        str
+    ] = get_fully_qualified_parameter_names(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+    )
+
+    """
+    These assertions show that dependent "ParameterBuilder" computed its values.
+    """
+    assert (
+        my_dependent_parameter_builder.raw_fully_qualified_parameter_name
+        in all_fully_qualified_parameter_names
+        or my_dependent_parameter_builder.json_serialized_fully_qualified_parameter_name
+        in all_fully_qualified_parameter_names
+    )
```

Comment:
Why is there an `or` here? Don't we know where the parameters are going to go?

Reply:
@donaldheppner Fantastic catch -- this should be `AND`!  Thank you VERY much (fixed in next commit).